src/shellstrings.c

```diff
@@ -7,88 +7,331 @@
+    if (*color == true) {
+        printf(BOLDWHITE "\nLine:" RESET " %d:\n%s" , lnum, line);
+
+        if (returnflag) {
+             printf("\n");
+        }
+
+        printf(BOLDRED "%s" RESET "\n", underline);
+    }
+
+    else {
+        printf("\nLine: %d:\n%s", lnum, line);
+
+        if (returnflag) {
+             printf("\n");
+        }
+
+        printf("%s\n", underline);
+    }
+}
+
+
+void shell_interactive_replacements(char *word, char **sug, int flag, bool *color) {
+    int j = 0;
+
+    if (*color == true) {
+        if (flag == EXIT_FAILURE) {
+            sug[0] = word;
+            printf("\nNo suggestions have been generated for " BOLDWHITE "%s" RESET ".\n\n", word);
+            printf(BOLDWHITE "[d]" RESET " : Delete\n");
+            printf(BOLDWHITE "[i]" RESET " : Input\n");
+            printf(BOLDWHITE "[s]" RESET " : Skip\n\n");
+        }
+
+        else {
+            printf("\nPossible replacements for word %s are:\n\n", word);
+
+            while (sug[j] != NULL) {
+                printf("[%d] : %s\n", j + 1, sug[j]);
+                j++;
+            }
+
+            printf(BOLDWHITE "[d]" RESET " : Delete\n");
+            printf(BOLDWHITE "[i]" RESET " : Input\n");
+            printf(BOLDWHITE "[s]" RESET " : Skip\n\n");
+        }
+    }
+
+    else {
+        if (flag == EXIT_FAILURE) {
+            sug[0] = word;
+            printf("\nNo suggestions have been generated for %s.\n\n", word);
+            printf("[d] : Delete\n");
+            printf("[i] : Input\n");
+            printf("[s] : Skip\n\n");
+        }
+
+        else {
+            printf("\nPossible replacements for word %s are:\n\n", word);
+
+            while (sug[j] != NULL) {
+                printf("[%d] : %s\n", j + 1, sug[j]);
+                
+                j++;
+            }
+
+            printf( "[d] : Delete\n");
+            printf( "[i] : Input\n");
+            printf( "[s] : Skip\n\n");
+        }
+    }
+}
+
+void shell_verbose_chart(int lnum, char* line, char *misspelled, char **suggestions) {
+    // Prints the location and character
+    char* ptr = strstr(line, misspelled);
+    int charpos = 0;
+    charpos = ptr - line;
+
+    log_trace("shell_verbose_chart character position in line is %d", charpos);
+    
+    printf("%d:%d\t\t\t", lnum, charpos);
+    
     int ntab = 3 - (strlen(misspelled) / 8); // number of tabs
+    
+    if (ntab < 0) {
+        ntab = 1;
+    }
+    
     int j;
+    
     printf("%s", misspelled);
-    for (j = 0; j < ntab; j++) printf("\t");
+    
+    for (j = 0; j < ntab; j++) {
+        printf("\t");
+    }
 
     j = 0;
+
     while (suggestions[j] != NULL) {
         printf("%s", suggestions[j]);
-        if (suggestions[j + 1] != NULL) printf(", ");
-        j++;
 
+        if (suggestions[j + 1] != NULL) {
+            printf(", ");
+        }
+
+        j++;
     }
+
     printf("\n"); // print list of replacement
 }
 
+void shell_save_message(bool *color) {
```

Comment:
Why is this a `bool *` and not just a `bool`?

Reply:
changed!